### PR TITLE
Enhance and add the integration.

### PR DESCRIPTION
**Pull request by niceboygithub**
1. Can use Integration to add another LINE-bot
2. Flexible message format.
3. Update README.md

### CODE REVIEW

**Comment by osk2** [comment]
Hi, @niceboygithub 
Is this PR finish yet? It seems you still pushing more commits

**Comment by tsunglung** [comment]
No bug found now. 
The new improvement will be the next stage.

**Comment by osk2** [comment]
Just tested this PR, I followed steps in readme but no notify service had been added into HA after I completing config flow, is there anything I missed?

Also, from my point of view, there's no reason to add default config into `configuration.yaml` if we already leverage the power of config flow

**Comment by tsunglung** [comment]
The notify domain was not supported by using config_entry to setup.
It only support setup_platform now.
You still need to config one in configuration.yaml.
For the second, third one, you can use config flow to add.

**Comment by osk2** [comment]
@tsunglung Apologize for late reply

I still not able to create second notify service using config flow. Nothing show up in my service list after completing the flow

**Comment by tsunglung** [comment]
You did not get my point.
Notify sevice did not support config entry setup.
Use config entry to add another client_id.
Then user can to send messages to another client_id by specifying name in the data field.
Let me add more description in the READMe later.

**Comment by osk2** [comment]
Ah that makes sense! I thought config flow is used to create service 🤦🏼  (may confuse users too?)

I'll try again later and see if there's anything we can improve

File: custom_components/line_bot/notify.py [hunk 1]
```diff
@@ -5,56 +5,93 @@
...
-                   'Content-type': 'application/json'}
+        name = None
+        if kwargs.get("data"):
+            name = kwargs["data"].get("name", self._line_bot[CONF_NAME])
+        client_id = self._line_bot[CONF_CLIENT_ID]
+        access_token = self._line_bot[CONF_ACCESS_TOKEN]
 
-        payload = json.loads(message.encode('utf-8'))
-        payload['to'] = self.client_id
+        for i in self._hass.data[DOMAIN][LINE_BOT]:
```
Comment (osk2) [inline custom_components/line_bot/notify.py]: This will cause KeyError if user only setup notify service via configuration.yaml

**Review by osk2 — CHANGES_REQUESTED** [review]

**Pull request merged**